opentracing/noop.go

```diff
@@ -0,0 +1,62 @@
+// Noop implementations of the core opentracing interfaces.
+package opentracing
+
+import "golang.org/x/net/context"
+
+// noopContextID:
+type noopContextID struct{}
```

Comment:
for efficiency, it's best to instantiate these once and always return the same instances. I did a simple perf test in Python implementation the other day, the perf difference was about 20 times between a noop tracer and a regular tracer that had sampling rate of 0. 

https://github.com/uber-common/opentracing-go/blob/master/noop.go#L28